src/plugins/ui_feature/data_cleanup.ts

```diff
@@ -0,0 +1,209 @@
+          ),
+          this.chainValidations(this.checkNoColumnProvided, this.checkColumnsAreUnique)
+        );
+    }
+    return CommandResult.Success;
+  }
+
+  handle(cmd: Command) {
+    switch (cmd.type) {
+      case "REMOVE_DUPLICATES":
+        this.removeDuplicates(cmd.columns, cmd.hasHeader);
+        break;
+    }
+  }
+
+  // ---------------------------------------------------------------------------
+  // Private
+  // ---------------------------------------------------------------------------
+
+  private removeDuplicates(columnsToAnalyze: HeaderIndex[], hasHeader: boolean) {
+    const sheetId = this.getters.getActiveSheetId();
+    const zone = this.getters.getSelectedZone();
+    if (hasHeader) {
+      zone.top += 1;
+    }
+
+    const uniqueRowsIndexes = this.getUniqueRowsIndexes(
+      sheetId,
+      zone.top,
+      zone.bottom,
+      columnsToAnalyze
+    );
+    const numberOfUniqueRows = uniqueRowsIndexes.length;
+
+    if (numberOfUniqueRows === zoneToDimension(zone).numberOfRows) {
+      this.notifyRowsRemovedAndRemaining(0, numberOfUniqueRows);
+      return;
+    }
+
+    const rowsToKeep: Zone[] = uniqueRowsIndexes.map((rowIndex) => ({
+      left: zone.left,
+      top: rowIndex,
+      right: zone.right,
+      bottom: rowIndex,
+    }));
+
+    const state = new ClipboardCellsState(
+      rowsToKeep,
+      "COPY",
+      this.getters,
+      this.dispatch,
+      this.selection
+    );
+
+    for (const { col, row } of positions(zone)) {
+      this.dispatch("CLEAR_CELL", { col, row, sheetId });
+    }
+
+    const zonePasted: Zone = {
+      left: zone.left,
+      top: zone.top,
+      right: zone.left,
+      bottom: zone.top,
+    };
+
+    state.paste([zonePasted]);
+
+    const remainingZone = {
+      left: zone.left,
+      top: zone.top - (hasHeader ? 1 : 0),
+      right: zone.right,
+      bottom: zone.top + numberOfUniqueRows - 1,
+    };
+
+    this.selection.selectZone({
+      cell: { col: remainingZone.left, row: remainingZone.top },
+      zone: remainingZone,
+    });
+
+    const removedRows = zone.bottom - zone.top + 1 - numberOfUniqueRows;
+    this.notifyRowsRemovedAndRemaining(removedRows, numberOfUniqueRows);
+  }
+
+  private getUniqueRowsIndexes(
+    sheetId: UID,
+    top: HeaderIndex,
+    bottom: HeaderIndex,
+    columns: HeaderIndex[]
+  ): number[] {
+    const uniqueRows: CellValueByRow = {};
+
+    for (const row of range(top, bottom + 1)) {
+      const cellsValuesInRow = columns.map((col) => {
+        return this.getters.getEvaluatedCell({
+          sheetId,
+          col,
+          row,
+        }).value;
+      });
+
+      const isRowUnique = !Object.values(uniqueRows).some((uniqueRow) =>
+        deepEquals(uniqueRow, cellsValuesInRow)
+      );
+
+      if (isRowUnique) {
+        uniqueRows[row] = cellsValuesInRow;
+      }
+    }
+    // transform key object in number
+    return Object.keys(uniqueRows).map((key) => parseInt(key));
```

Comment:
If you use a map I think you could simply do `Array.from(uniquesRows.keys())` and it will keep the correct type (in JS it seems to be the case)

Reply:
Change taken into account :)